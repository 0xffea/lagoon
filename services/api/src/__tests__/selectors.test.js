// @flow


const { getAllSiteGroups } = require('../selectors'); 


describe('getAllSiteGroups', () => {
  test('should return a list of getAllSiteGroups', () => {
    const state = {
      siteGroups: {
        sg1: { gitUrl: 'git://sg1' },
        sg2: { gitUrl: 'git://sg2' },
      },
    };

    const ret = getAllSiteGroups(state);

    expect(ret).toMatchSnapshot();
  });
});
