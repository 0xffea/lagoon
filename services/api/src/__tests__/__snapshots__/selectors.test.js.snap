// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getAllSiteGroups should return a complete list of siteGroups 1`] = `
Array [
  Object {
    "gitUrl": "git://sg1",
    "id": "sg1",
    "site_group_name": "sg1",
  },
  Object {
    "gitUrl": "git://sg2",
    "id": "sg2",
    "site_group_name": "sg2",
  },
]
`;

exports[`getAllSitesByEnv should return a complete list of sites 1`] = `
Array [
  Object {
    "computed": Object {
      "fileName": "compact/deploytest1.yaml",
      "id": "deploytest1.compact/deploytest_branch1",
      "jumpHost": undefined,
      "serverIdentifier": "deploytest1",
      "serverInfrastructure": "compact",
      "serverNames": Array [
        "deploytest1.compact",
      ],
      "siteHost": "deploytest1.compact",
      "siteName": "deploytest_branch1",
    },
    "site_branch": "branch1",
    "site_environment": "development",
    "sitegroup": "deploytest",
    "uid": 3201,
  },
  Object {
    "computed": Object {
      "fileName": "compact/deploytest2.yaml",
      "id": "deploytest2.compact/deploytest_branch1",
      "jumpHost": undefined,
      "serverIdentifier": "deploytest2",
      "serverInfrastructure": "compact",
      "serverNames": Array [
        "deploytest2.compact",
      ],
      "siteHost": "deploytest2.compact",
      "siteName": "deploytest_branch1",
    },
    "site_branch": "branch2",
    "site_environment": "development",
    "sitegroup": "deploytest",
    "uid": 3201,
  },
]
`;
